index.bs

```diff
@@ -3481,6 +3638,309 @@ partial dictionary MLOpSupportLimits {
+    **Arguments:**
+        - <dfn>input</dfn>: an {{MLOperand}}. The input tensor.
+        - <dfn>scale</dfn>: an {{MLOperand}}. The scale tensor to divide each input value before adjusting by the zero point. It must be [=blockwise broadcastable=] with the input.
+        - <dfn>zeroPoint</dfn>: an {{MLOperand}}. The zero point tensor to add to each rescaled input value. It has the same [=MLOperand/shape=] as the scale.
+        - <dfn>options</dfn>: an {{MLOperatorOptions}}. Specifies the optional parameters of the operation.
+
+    **Returns:** an {{MLOperand}}. The output tensor that contains the quantized values.
+</div>
+
+<table id=constraints-quantizelinear class='data' link-for="MLGraphBuilder/quantizeLinear(input, scale, zeroPoint, options)">
+  <caption>Constraints for {{MLGraphBuilder/quantizeLinear()}}</caption>
+  <thead>
+    <tr>
+      <th>operand</th>
+      <th>[=/allowed data types=]</th>
+      <th>[=/allowed ranks=]</th>
+    </tr>
+  </thead>
+  <tr>
+    <td>{{input}}</td>
+    <td>{{MLOperandDataType/"float32"}}, {{MLOperandDataType/"float16"}}</td>
+    <td>[=/any rank|N=]</td>
+  </tr>
+  <tr>
+    <td>{{scale}}</td>
+    <td>[=/same type as|same as=] {{input}}</td>
+    <td>[=/same rank as|same as=] {{input}}</td>
+  </tr>
+  <tr>
+    <td>{{zeroPoint}}</td>
+    <td>{{MLOperandDataType/"uint8"}}, {{MLOperandDataType/"uint32"}}, {{MLOperandDataType/"int32"}}</td>
+    <td>[=/same rank as|same as=] {{input}}</td>
+  </tr>
+  <tr>
+    <td>*output*</td>
+    <td>[=/same type as|same as=] {{zeroPoint}}</td>
+    <td>[=/same rank as|same as=] {{input}}</td>
+  </tr>
+</table>
+
+{{MLOpSupportLimits}} has the following member for {{MLGraphBuilder/quantizeLinear()}}:
+<dl dfn-type=dict-member dfn-for=MLOpSupportLimits>
+    : <dfn>quantizeLinear</dfn>
+    :: Support limits for operator {{MLGraphBuilder/quantizeLinear()}}.
+</dl>
+
+
+<details open algorithm>
+  <summary>
+    The <dfn method for=MLGraphBuilder>quantizeLinear(|input|, |scale|, |zeroPoint|, |options|)</dfn> method steps are:
+  </summary>
+    1. If [=this=].{{MLGraphBuilder/[[hasBuilt]]}} is true, then [=exception/throw=] an "{{InvalidStateError}}" {{DOMException}}.
+    1. If [=MLGraphBuilder/validating operand=] with [=this=] and any of |input|, |scale|, and |zeroPoint| returns false, then [=exception/throw=] a {{TypeError}}.
+    1. If |scale|'s [=MLOperand/rank=] or |zeroPoint|'s [=MLOperand/rank=] is not equal to |input|'s [=MLOperand/rank=], then [=exception/throw=] a {{TypeError}}.
+    1. If |scale|'s [=MLOperand/shape=] is not equal to |zeroPoint|'s [=MLOperand/shape=], then [=exception/throw=] a {{TypeError}}.
+    1. If [=blockwise broadcasting=] |scale|'s [=MLOperand/shape=] and |input|'s [=MLOperand/shape=] returns false, then [=exception/throw=] a {{TypeError}}.
+    1. If [=blockwise broadcasting=] |zeroPoints|'s [=MLOperand/shape=] and |input|'s [=MLOperand/shape=] returns false, then [=exception/throw=] a {{TypeError}}.
```

Comment:
```suggestion
    1. If [=blockwise broadcasting=] |zeroPoint|'s [=MLOperand/shape=] and |input|'s [=MLOperand/shape=] returns false, then [=exception/throw=] a {{TypeError}}.
```

Reply:
Is this line even necessary as _scale_ and _zeroPoint_ must have the same shape?

Reply:
> 1. If [=blockwise broadcasting=] |zeroPoint|'s [=MLOperand

True, technically not. From the POV of clarity for operator behavior and contracts, this redundancy highlights that scale and zeroPoint must be *both* broadcastable to the input; and arguably the reverse is true too, that we could instead delete the line *above* it checking `scale`. I think it's okay for an implementation to have a more efficient implementation or skip any algorithm steps so long as they produce the same end result the steps would have yielded.

Reply:
No objections to keeping it!